### PR TITLE
FIX: Skip version cache on read-only filesystems

### LINKED ISSUE #853: Difficulty running singularity image of 0.15.2:  Read-only file system: '/home/bidsapp/.cache/mriqc'
I am not sure if it is a problem in mriqc docker image itself, or may be it is due to the difference in singularity version used to produce 0.15.1 and 0.15.2 images (from docker images) for https://github.com/ReproNim/containers ... but with 0.15.2, if I follow our https://github.com/ReproNim/containers#a-typical-workflow (just remove/comment out the freezing to 0.15.1 so you would get 0.15.2) I would get

```shell
> datalad containers-run -n containers/bids-mriqc --input sourcedata --output . '{inputs}' '{outputs}' participant group
[INFO   ] Making sure inputs are available (this may take some time)
get(ok): /home/yoh/.tmp/repro-KtxiqAt/ds000003-qc/sourcedata/sub-02/anat/sub-02_T1w.nii.gz (file) [from s3-PUBLIC...]
get(ok): /home/yoh/.tmp/repro-KtxiqAt/ds000003-qc/sourcedata/sub-02/anat/sub-02_inplaneT2.nii.gz (file) [from s3-PUBLIC...]
get(ok): /home/yoh/.tmp/repro-KtxiqAt/ds000003-qc/sourcedata/sub-02/func/sub-02_task-rhymejudgment_bold.nii.gz (file) [from s3-PUBLIC...]
get(ok): /home/yoh/.tmp/repro-KtxiqAt/ds000003-qc/sourcedata/sub-13/anat/sub-13_T1w.nii.gz (file) [from s3-PUBLIC...]
get(ok): /home/yoh/.tmp/repro-KtxiqAt/ds000003-qc/sourcedata/sub-13/anat/sub-13_inplaneT2.nii.gz (file) [from s3-PUBLIC...]
get(ok): /home/yoh/.tmp/repro-KtxiqAt/ds000003-qc/sourcedata/sub-13/func/sub-13_task-rhymejudgment_bold.nii.gz (file) [from s3-PUBLIC...]
get(ok): /home/yoh/.tmp/repro-KtxiqAt/ds000003-qc/containers/images/bids/bids-mriqc--0.15.2.sing (file) [from origin...]
[INFO   ] == Command start (output follows) =====
Traceback (most recent call last):
  File "/usr/local/miniconda/bin/mriqc", line 10, in <module>
    sys.exit(main())
  File "/usr/local/miniconda/lib/python3.7/site-packages/mriqc/cli/run.py", line 15, in main
    parse_args()
  File "/usr/local/miniconda/lib/python3.7/site-packages/mriqc/cli/parser.py", line 368, in parse_args
    parser = _build_parser()
  File "/usr/local/miniconda/lib/python3.7/site-packages/mriqc/cli/parser.py", line 340, in _build_parser
    latest = check_latest()
  File "/usr/local/miniconda/lib/python3.7/site-packages/mriqc/cli/version.py", line 22, in check_latest
    cachefile.parent.mkdir(parents=True, exist_ok=True)
  File "/usr/local/miniconda/lib/python3.7/pathlib.py", line 1241, in mkdir
    self._accessor.mkdir(self, mode)
OSError: [Errno 30] Read-only file system: '/home/bidsapp/.cache/mriqc'
[INFO   ] == Command exit (modification check follows) =====
[INFO   ] The command had a non-zero exit code. If this is expected, you can save the changes with 'datalad save -d . -r -F .git/COMMIT_EDITMSG'
CommandError: ''containers/scripts/singularity_cmd run containers/images/bids/bids-mriqc--0.15.2.sing '"'"'sourcedata'"'"' '"'"''"'"' participant group'' failed with exitcode 1 under /home/yoh/.tmp/repro-KtxiqAt/ds000003-qc
```

I have not really figured out **why** 0.15.2 specifically falls into the trap, and how `~/` inside the container is not the one we prepare/provide to singularity invocation (in [here](https://github.com/ReproNim/containers/blob/master/scripts/singularity_cmd#L82)) or is it the singularity version (3.5.2+ds1-1~ndall+1) to matter.

May be there is some difference in version of singularity which was used to build those two versions awhile back, but their recipes are nearly identical:
```
(git)smaug:~/proj/repronim/containers[master]git
$> diff images/bids/Singularity.bids-mriqc--0.15.{1,2}
6c6
< From: poldracklab/mriqc:0.15.1
---
> From: poldracklab/mriqc:0.15.2
15,16c15,16
< export LANG="en_US.UTF-8"
< export LC_ALL="en_US.UTF-8"
---
> export LANG="C.UTF-8"
> export LC_ALL="C.UTF-8"
```

So just decided to reach out may be for bright ideas/memories of possible changes between 0.15.1 and 0.15.2 which could attribute here (I will test later more by rebuilding both singularity images with consistent/same version etc)

Relates in spirit to an elderly #709 ;-)

### PR DESCRIPTION
**Pull request by effigies**
Fixes #853.

Following nipreps/fmriprep#1830.

### CODE REVIEW

File: mriqc/cli/version.py [hunk 1]
```diff
@@ -19,21 +19,25 @@ def check_latest():
     date = None
     outdated = None
     cachefile = Path.home() / ".cache" / "mriqc" / "latest"
-    cachefile.parent.mkdir(parents=True, exist_ok=True)
+        try:
```
Comment (effigies) [inline mriqc/cli/version.py:22]:
```suggestion
    try:
```

**Review by oesteban — APPROVED** [review]

**Comment by effigies** [comment]
Travis failures are #863. Will not include fix in this PR.

**Comment by effigies** [comment]
@oesteban There are Bad Gateway errors in the CircleCI tests. URLs aren't exposed, so I don't know what service it's trying to access.

cc @rwblair in case you know and can restart a service.

**Comment by rwblair** [comment]
Looks like its running its own version of the webapi locally:
https://github.com/poldracklab/mriqc/blob/5898396026817ce114bbc90bac0dd5d7c48618e0/.circleci/config.yml#L468

Guess I'll rerun job with ssh and inspect nginx logs to see what I can see.

**Comment by effigies** [comment]
Did we ever figure this out?

**Comment by effigies** [comment]
SSHing in, I see:

```
$ docker logs dockereve-master_eve_1
[2020-12-13 23:42:24 +0000] [1] [INFO] Starting gunicorn 20.0.4
[2020-12-13 23:42:24 +0000] [1] [INFO] Listening at: http://0.0.0.0:5000 (1)
[2020-12-13 23:42:24 +0000] [1] [INFO] Using worker: sync
[2020-12-13 23:42:24 +0000] [8] [INFO] Booting worker with pid: 8
[2020-12-13 23:42:24 +0000] [9] [INFO] Booting worker with pid: 9
[2020-12-13 23:42:25 +0000] [8] [ERROR] Exception in worker process
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/gunicorn/arbiter.py", line 583, in spawn_worker
    worker.init_process()
  File "/usr/local/lib/python3.6/site-packages/gunicorn/workers/base.py", line 119, in init_process
    self.load_wsgi()
  File "/usr/local/lib/python3.6/site-packages/gunicorn/workers/base.py", line 144, in load_wsgi
    self.wsgi = self.app.wsgi()
  File "/usr/local/lib/python3.6/site-packages/gunicorn/app/base.py", line 67, in wsgi
    self.callable = self.load()
  File "/usr/local/lib/python3.6/site-packages/gunicorn/app/wsgiapp.py", line 49, in load
    return self.load_wsgiapp()
  File "/usr/local/lib/python3.6/site-packages/gunicorn/app/wsgiapp.py", line 39, in load_wsgiapp
    return util.import_app(self.app_uri)
  File "/usr/local/lib/python3.6/site-packages/gunicorn/util.py", line 358, in import_app
    mod = importlib.import_module(module)
  File "/usr/local/lib/python3.6/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 994, in _gcd_import
  File "<frozen importlib._bootstrap>", line 971, in _find_and_load
  File "<frozen importlib._bootstrap>", line 955, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/usr/src/app/app.py", line 21, in <module>
    app.register_blueprint(swagger, url_prefix='/docs/api')
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 98, in wrapper_func
    return f(self, *args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1156, in register_blueprint
    if blueprint.name in self.blueprints:
AttributeError: module 'eve_swagger.swagger' has no attribute 'name'
[2020-12-13 23:42:25 +0000] [8] [INFO] Worker exiting (pid: 8)
[2020-12-13 23:42:25 +0000] [9] [ERROR] Exception in worker process
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/gunicorn/arbiter.py", line 583, in spawn_worker
    worker.init_process()
  File "/usr/local/lib/python3.6/site-packages/gunicorn/workers/base.py", line 119, in init_process
    self.load_wsgi()
  File "/usr/local/lib/python3.6/site-packages/gunicorn/workers/base.py", line 144, in load_wsgi
    self.wsgi = self.app.wsgi()
  File "/usr/local/lib/python3.6/site-packages/gunicorn/app/base.py", line 67, in wsgi
    self.callable = self.load()
  File "/usr/local/lib/python3.6/site-packages/gunicorn/app/wsgiapp.py", line 49, in load
    return self.load_wsgiapp()
  File "/usr/local/lib/python3.6/site-packages/gunicorn/app/wsgiapp.py", line 39, in load_wsgiapp
    return util.import_app(self.app_uri)
  File "/usr/local/lib/python3.6/site-packages/gunicorn/util.py", line 358, in import_app
    mod = importlib.import_module(module)
  File "/usr/local/lib/python3.6/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 994, in _gcd_import
  File "<frozen importlib._bootstrap>", line 971, in _find_and_load
  File "<frozen importlib._bootstrap>", line 955, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/usr/src/app/app.py", line 21, in <module>
    app.register_blueprint(swagger, url_prefix='/docs/api')
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 98, in wrapper_func
    return f(self, *args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1156, in register_blueprint
    if blueprint.name in self.blueprints:
AttributeError: module 'eve_swagger.swagger' has no attribute 'name'
[2020-12-13 23:42:25 +0000] [9] [INFO] Worker exiting (pid: 9)
[2020-12-13 23:42:25 +0000] [1] [INFO] Shutting down: Master
[2020-12-13 23:42:25 +0000] [1] [INFO] Reason: Worker failed to boot.
```

(Message repeats...)

Did some digging, and came up with https://github.com/poldracklab/mriqcwebapi/pull/47.

Testing that here.

**Comment by effigies** [comment]
IQM issue. @oesteban if you could have a look? Or anybody else qualified to look through Circle?

**Comment by oesteban** [comment]
Could you rebase?

**Comment by effigies** [comment]
Tests passing and I got an in-principle approval a while back, so I'll go ahead and merge.

I would suggest a release soonish since a lot of these issues are long-standing, though we might also need to release mriqcwebapi. I don't know if there are deployment issues there?

**Comment by yarikoptic** [comment]
note to myself -- slated for 0.15.3 or 0.16 whatever comes first, right? ;)

**Comment by oesteban** [comment]
> I don't know if there are deployment issues there?

You bet, but maybe the webapi wants to give us a nice surprise. @rwblair may actually know better than myself

**Comment by effigies** [comment]
0.15.3 is out already. I would suggest 0.16, just to mark that it took some work and upgraders may need to consider dependencies.

**Comment by yarikoptic** [comment]
https://github.com/poldracklab/mriqc/ says latest release 0.15.2

**Comment by effigies** [comment]
There's a tag that didn't get an official release page. If you click in the releases, you'll see it.

**Pull request merged**